src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -405,14 +380,18 @@ internal static Microsoft.CodeAnalysis.Emit.AnonymousTypeKey GetAnonymousTypeKey
         internal IReadOnlyDictionary<Microsoft.CodeAnalysis.Emit.AnonymousTypeKey, Microsoft.CodeAnalysis.Emit.AnonymousTypeValue> GetAnonymousTypeMap()
         {
             var result = new Dictionary<Microsoft.CodeAnalysis.Emit.AnonymousTypeKey, Microsoft.CodeAnalysis.Emit.AnonymousTypeValue>();
-            var templates = GetAllCreatedTemplates();
-            foreach (AnonymousTypeTemplateSymbol template in templates)
+            var templates = ArrayBuilder<AnonymousTypeTemplateSymbol>.GetInstance();
+            // Get anonymous types but not synthesized delegates
+            // since delegate types are not reused across generations.
```

Comment:
No, it's just additional complexity. To reuse delegates from the initial generation (loaded from metadata) we'd need to parse the mangled delegate metadata names for instance.